src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -758,9 +758,20 @@ static int rbd_nbd(int argc, const char *argv[])
 
   argv_to_vec(argc, argv, args);
   env_to_vec(args);
-  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT,
-			 CODE_ENVIRONMENT_DAEMON,
-			 CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS);
+
+  code_environment_t code_env = CODE_ENVIRONMENT_UTILITY;
+  int env_flag = 0;
+  for (auto arg : args) {
+    if (arg != NULL && arg[0] != '-') {
+      if (strcmp(arg, "map") == 0) {
+        code_env = CODE_ENVIRONMENT_DAEMON;
+        env_flag = CINIT_FLAG_UNPRIVILEGED_DAEMON_DEFAULTS;
+      }
+      break;
+    }
+  }
+
+  auto cct = global_init(NULL, args, CEPH_ENTITY_TYPE_CLIENT, code_env, env_flag);
```

Comment:
> BTW, our rbd util is also affected: it creates admin socket file when is running.

Also interesting effect found during testing: if a socket is created by root but in a user directory, and then a rbd commad is run by the user, it removes the root's asok, because before binding, it checks if the socket is alive (and this fails when testing a root socket by a non-root), and it removes this socket as dead.

Yes! I also found this issue before, the only way to resolve it I can imagine is to make the asok file name with nbd index, or process id... What is your opinion?